lib/manageiq/rpm_build/nightly_build_purger.rb

```diff
@@ -0,0 +1,53 @@
+module ManageIQ
+  module RPMBuild
+    class NightlyBuildPurger
+      include S3Common
+
+      def run
+        candidates = {}
+        keepers    = {}
+
+        client.list_objects(:bucket => OPTIONS.rpm_repository.s3_api.bucket, :prefix => File.join("builds", "manageiq-nightly")).flat_map(&:contents).each do |object|
+          package, timestamp    = package_timestamp_from_key(object.key)
+          candidates[package] ||= []
+          keepers[package]    ||= []
+
+          if recent?(timestamp)
+            keepers[package] << object.key
+            next
+          end
+
+          position = (timestamp.to_i - 20_000_000_000_000) / 10_000 # YYMMDDHH i.e. 22091300
```

Comment:
I assume the format is yyyymmddhhmmss ? If so, this might be easier to read:

```suggestion
          position = timestamp[2, 8] # YYYYMMDDHHMMSS -> YYMMDDHH
```

Reply:
Though I don't understand what this "position" means

Reply:
Done